common/filetoolkitwithundo.h

```diff
@@ -127,14 +109,9 @@ class FileToolkitWithUndo
      * @brief m_cleanupHandlers A vector of cleanupHandlers added during the lifetime of the
      *  FileToolKitWithUndo that will be run from the destructor.
      */
-    std::vector<CleanUpHandler *> m_cleanupHandlers;
-private :
-    /**
-     * @brief createParentDirectory Recursively tries to create the directory pointed to by path.
-     * @param path The directory path to be created.
-     * @return true on success, false on failure
-     */
-    bool createParentDirectory(const std::string &path);
+    std::vector<std::unique_ptr<CleanUpHandler>> m_cleanupHandlers;
+
+    CreateDir m_create;
```

Comment:
Separating the directory creation code like this feels unnatural, and causes code duplication later on (either clean, or add the stuff done with m_create to a cleanuphandler anyway)